### PR TITLE
prevent pasting a node after modifying the copied node

### PR DESCRIPTION
**Pull request by anas-drissi**

### CODE REVIEW

File: src/translations/fr.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
     "moveStudyNotification": "L'étude a été déplacée de {oldStudyPath} à {studyPath}",
 
+    "invalidateCopyPastOfNode": "Invalidation du collage du noeud suite à la modification du neoud copié",
```
Comment (sBouzols) [inline src/translations/fr.json:20]: invalidateCopy~~Past~~OfNode
I propose this text :
"Invalidation de la copie du nœud suite à modification"

**Review by sBouzols — CHANGES_REQUESTED** [review]
take care with effects

File: src/components/network-modification-tree-pane.js [hunk 2]
```diff
@@ -156,6 +156,24 @@ export const NetworkModificationTreePane = ({
...
+                    studyUpdatedForce.eventData.headers['nodes'].some(
+                        (nodeId) => nodeId === selectedNodeIdForCopy
+                    )
+                ) {
+                    setSelectedNodeIdForCopy(null);
+                    snackInfo('', 'invalidateCopyPastOfNode');
+                }
+            }
+        }
+    }, [studyUpdatedForce, updateNodes, selectedNodeIdForCopy, snackInfo]);
```
[the remark this answers is not in the document]
Comment (anas-drissi) [reply src/components/network-modification-tree-pane.js:175]: The useEffect should fire when the list of copied nodes will change.

File: src/translations/fr.json (hunk 1 above)
Comment (anas-drissi) [reply src/translations/fr.json:20]: It's fixed

File: src/components/network-modification-tree-pane.js [hunk 3]
```diff
@@ -152,9 +152,27 @@ export const NetworkModificationTreePane = ({
...
                     );
                 }
+                if (
+                    studyUpdatedForce.eventData.headers['nodes'].some(
+                        (nodeId) => nodeId === selectedNodeIdForCopy
+                    )
+                ) {
+                    setSelectedNodeIdForCopy(null);
+                    snackInfo(
+                        '',
```
Comment (SlimaneAmar) [inline src/components/network-modification-tree-pane.js:162]: Add a message key not the message
Add the message content param not the header param

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/translations/fr.json [hunk 4]
```diff
@@ -17,6 +17,7 @@
 
     "moveStudyNotification": "L'étude a été déplacée de {oldStudyPath} à {studyPath}",
 
+    "Invalidation de la copie du nœud suite à modification": "Invalidation du collage du noeud suite à la modification du neoud copié",
```
Comment (sBouzols) [inline src/translations/fr.json:20]: validated with Pauline :

"CopiedNodeInvalidationMessage": "Suite à modification de nœud, le contenu du presse papier a été supprimé"

File: src/translations/en.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
 
     "moveStudyNotification": "The study has been moved from {oldStudyPath} to {studyPath}",
 
+    "Invalidation de la copie du nœud suite à modification": "Invalidation of pasting of the node following the modification of the copied node",
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/translations/en.json:20]: validated with Pauline :

"CopiedNodeInvalidationMessage": "As a result of node modification, the contents of the clipboard have been deleted"

File: src/components/network-modification-tree-pane.js [hunk 6]
```diff
@@ -152,9 +152,27 @@ export const NetworkModificationTreePane = ({
...
+                }
             }
         }
-    }, [studyUuid, studyUpdatedForce, updateNodes, dispatch]);
+    }, [
+        studyUuid,
+        studyUpdatedForce,
+        updateNodes,
+        dispatch,
+        selectedNodeIdForCopy,
```
Comment (sBouzols) [inline src/components/network-modification-tree-pane.js:173]: You will still fire this effect when the copied node id changes.
This is not what we expect.
This effect should only fire when there is a new notification to consume.

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/network-modification-tree-pane.js (hunk 3 above)
Comment (anas-drissi) [reply src/components/network-modification-tree-pane.js:162]: See with @sBouzols 
Every where in existing code the first param of snackInfo, snackError, snackWarning is a not translated error message which mainly come from back services.
Only the header message could use a translation key (headerMEssageId) to compute translation.
Maybe we could create another task to make this evolution to add a translated text in the content body of the snackbar.

**Review by sBouzols — APPROVED** [review]
code Review OK
Test OK

File: src/utils/messages.js [hunk 7]
```diff
@@ -31,18 +31,15 @@ function displayMessageWithSnackbar({
     level,
     persistent,
 }) {
-    let message;
+    let message = intlRef.current.formatMessage({ id: errorMessage });
```
Comment (SlimaneAmar) [inline src/utils/messages.js:34]: Remove these modifications

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/utils/messages.js (hunk 7 above)
Comment (anas-drissi) [reply src/utils/messages.js:34]: We will have more message in the scnakbar.

Comment (anas-drissi) [reply src/utils/messages.js]: It's fixed.

File: src/components/network-modification-tree-pane.js [hunk 8]
```diff
@@ -81,12 +81,15 @@ export const NetworkModificationTreePane = ({
...
     const intlRef = useIntlRef();
     const { enqueueSnackbar } = useSnackbar();
-    const { snackError } = useSnackMessage();
+    const { snackError, snackInfo } = useSnackMessage();
     const classes = useStyles();
     const DownloadIframe = 'downloadIframe';
 
     const [activeNode, setActiveNode] = useState(null);
     const [selectedNodeIdForCopy, setSelectedNodeIdForCopy] = useState(null);
+    const selectedNodeIdForCopyRef = useRef();
```
Comment (SlimaneAmar) [inline src/components/network-modification-tree-pane.js:90]: use init value useRef(selectedNodeIdForCopy);

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
When a node not buit and copied the copy is invalidated when we build the node
Fix will be made with the botifications refactoring

**Pull request merged**